src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs

```diff
@@ -969,11 +978,19 @@ internal static void ChoosePunctuators(ILanguageSettings loc, out string dec, ou
 
         private sealed class LexerImpl
         {
+            public enum LexerMode
+            {
+                Normal,
+                Island,
```

Comment:
>Island [](http://example.com/codeflow?start=16&length=6)

Can you commend what these modes mean? Especially the difference between Island and StringInterpolation?

Reply:
Added a comment.